polars/polars-core/src/series/mod.rs

```diff
@@ -593,7 +593,7 @@ pub trait SeriesTrait:
     }
 
     /// Retrieve the indexes needed for a sort.
-    fn argsort(&self, _reverse: bool) -> UInt32Chunked {
+    fn argsort(&self, _reverse: bool) -> Int32Chunked {
```

Comment:
This can be problematic. Polars uses `u32` as index type. This allows for 2^32 rows. By using signed integers, we half the number of rows available in a Polars `Series`/  `DataFrame`.

Reply:
I agree. if size is a concern, shouldn't we use u64/i64, then?

Reply:
we are already using LargeUtf8 to support large arrays, right?

Reply:
> I agree. if size is a concern, shouldn't we use u64/i64, then?

Yes, but this has a very noticable difference in cache-coherence. 

A future plan is to choose your index type with a feature flag and release a `polars[large-index]`  for python or something like that.

Reply:
> we are already using LargeUtf8 to support large arrays, right?

Yes, but I wish there was a `Utf8Array<u32>` :smile:

Reply:
got it. Let's add `Index` to the `take` in arrow2 👍